govcd/vm_test.go

```diff
@@ -652,6 +652,48 @@ func (vcd *TestVCD) Test_AnswerVmQuestion(check *C) {
 	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, false)
 }
 
+func (vcd *TestVCD) Test_VMChangeRootDiskSize(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+	vapp := vcd.findFirstVapp()
+	existingVm, vmName := vcd.findFirstVm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+
+	currentDisk := 0
+	if nil != vcd.vapp.VApp.Children.VM[0] && nil != vcd.vapp.VApp.Children.VM[0].VirtualHardwareSection && nil != vcd.vapp.VApp.Children.VM[0].VirtualHardwareSection.Item {
+		for _, item := range vcd.vapp.VApp.Children.VM[0].VirtualHardwareSection.Item {
```

Comment:
I'd like to see some comments explaining what are the expectations of this test, or what prerequisites we should meet to make the test pass.
When I run it, I get a failure when it expects to see `currentDisk` containing a non-zero value.

Reply:
This may be a consequence of the OVA used to construct the VM. I'm manually testing this with an Ubuntu OVA with a 10GB root disk, which I can then modify to be larger. If you could provide more information about the error and any disks attached to the test OVA, that would be helpful.

Reply:
What we use is a standard Photon OVA (http://dl.bintray.com/vmware/photon/3.0/GA/ova/:photon-hw11-3.0-26156e2.ova). 

```
START: vm_test.go:655: TestVCD.Test_VMChangeRootDiskSize
vm_test.go:674:
    check.Assert(0, Not(Equals), currentDisk)
... obtained int = 0
... expected int = 0

FAIL: vm_test.go:655: TestVCD.Test_VMChangeRootDiskSize
```

Reply:
Thanks. I'll get a VM spun up with that to test against.

Reply:
You can look at `createTestVapp` in api_vcd_test.go to see how the initial testing vApp is created for our suite.